### PR TITLE
[Shapes] Add percentage API to MDCCornerTreatment

### PR DESCRIPTION
**Pull request by yarneo**
This is a follow up PR to #5090 and contains all of its changes as well until that PR is merged.

We are adding a percentage API to MDCCornerTreatment by adding the removed value type from MDCShapeCorner and moving it into MDCCornerTreatment. This allows us to provide percentage values to corners on a component level and let it be set in the right time in the lifecycle.

Percentage support for each component will be added as follow up PRs once this is landed.

### CODE REVIEW

File: components/private/ShapeLibrary/src/MDCCornerTreatment+CornerTypeInitalizer.h [hunk 1]
```diff
@@ -28,6 +28,17 @@
...
+/**
+ Initialize and return an MDCCornerTreatment as an MDCRoundedCornerTreatment.
+
+ @param value The radius to set the rounded corner to.
+ @param valueType The value type in which the value is set as. It can be sent either as an
+ absolute value, or a percentage value (0.0 - 1.0) of the height of the surface.
+ @return an MDCRoundedCornerTreatment.
+ */
++ (MDCRoundedCornerTreatment *)cornerWithRadius:(CGFloat)value
+                                   andValueType:(MDCCornerTreatmentValueType)valueType;
```
Comment (jverkoey) [inline components/private/ShapeLibrary/src/MDCCornerTreatment+CornerTypeInitalizer.h]: Suggestion: drop the `and` from the API names - I think this is a relatively uncommon pattern in Objective-C APIs.

File: components/private/ShapeLibrary/src/MDCRoundedCornerTreatment.m [hunk 2]
```diff
@@ -58,6 +58,8 @@ - (MDCPathGenerator *)pathGeneratorForCornerWithAngle:(CGFloat)angle {
 - (BOOL)isEqual:(id)object {
   if (object == self) {
     return YES;
+  } else if (![super isEqual:object]) {
+    return NO;
```
Comment (jverkoey) [inline components/private/ShapeLibrary/src/MDCRoundedCornerTreatment.m:62]: Suggestion: break these changes out to a separate PR.

**Review by jverkoey — COMMENTED** [review]
Leaving initial feedback on what I think are the new changes.

**Review by jverkoey — APPROVED** [review]

File: components/private/ShapeLibrary/src/MDCRoundedCornerTreatment.m (hunk 2 above)
Comment (yarneo) [reply components/private/ShapeLibrary/src/MDCRoundedCornerTreatment.m:62]: there were actually mandatory for the tests to work, so thankfully because of tests i found a potential bug 🍬

Comment (jverkoey) [reply components/private/ShapeLibrary/src/MDCRoundedCornerTreatment.m:62]: Ah cool!
